### PR TITLE
Gradle reorganization and updates

### PR DESCRIPTION
**Pull request by TacoTheDank**
- Update Android Studio 3.0.1 -> 3.3.2
- Update Gradle 4.1 -> 4.10.3
- Move proguard files into a folder
- Replace buildscript repositories with their simpler names
- Replace compile with implementation
- Update buildtools 27.0.1 -> 28.0.3
- Delete redundant semicolon
- Set encoding to UTF-8

### Dependency updates
- Made everything generally more readable
- Support libraries 27.0.2 -> 27.1.1
- Constraint-layout 1.1.2 -> 1.1.3
- Jackson-databind 2.9.5 -> 2.9.8
- ShortcutBadger 1.1.16 -> 1.1.19
- PhotoView 2.1.3 -> 2.1.4
- Roundedimageview 2.1.0 -> 2.1.2
- Materialish-progress 1.5 -> 1.7
- Stickyheadergrid 0.9.4 -> 0.9.6
- All the test libraries moved to their latest version, as they aren't currently used and can be safely updated this way
- All dependency verifications updated accordingly (and redundant ones removed)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -204,17 +214,17 @@ android {
...
+                    'proguard/proguard-glide.pro',
+                    'proguard/proguard-jackson.pro',
+                    'proguard/proguard-retrofit.pro',
+                    'proguard/proguard-retrolambda.pro',
+                    'proguard/proguard-rounded-image-view.pro',
+                    'proguard/proguard-shortcutbadger.pro',
+                    'proguard/proguard.cfg'
+            testProguardFiles 'proguard/proguard-automation.pro',
+                    'proguard/proguard.cfg'
+            applicationIdSuffix ".debug"
```
Comment (cyBerta) [inline build.gradle]: Please leave the applicationIdSuffix as it was.

File: build.gradle [hunk 2]
```diff
@@ -1,160 +1,173 @@
...
-    testCompile 'org.powermock:powermock-api-mockito:1.6.1'
-    testCompile 'org.powermock:powermock-module-junit4:1.6.1'
-    testCompile 'org.powermock:powermock-module-junit4-rule:1.6.1'
-    testCompile 'org.powermock:powermock-classloading-xstream:1.6.1'
-
-    androidTestCompile ('org.assertj:assertj-core:1.7.1') {
+    })
+
+    /** Testing libraries */
+    testImplementation 'junit:junit:4.13-beta-2'
```
Comment (cyBerta) [inline build.gradle]: 4.12 should be sufficient, it's the latest stable version

**Comment by cyBerta** [comment]
Hi @TacoTheDank, thanks for your contribution!
Could you please rebase our master into your PR? Moreover, it would be cool to check if the tests from upstream master are running and I've added some comments.
Cheers

**Comment by TacoTheDank** [comment]
@cyBerta There we go :)

**Comment by cyBerta** [comment]
Hi @TacoTheDank we do have at least some tests now so the test libraries are used now. Moreover I would like to prefer using stable versions instead of beta versions for our dependencies. Anyway, I will fork your work, adapt it accordingly and merge it master after the next release. Thanks for your initiative to update the deps :)

**Comment by TacoTheDank** [comment]
Yep :)

**Pull request closed without merging**